SEAPODYM_Data_Test.py

```diff
@@ -0,0 +1,94 @@
+                   help='List of field variables to extract, using PARCELS naming convention')
+    p.add_argument('-n', '--netcdf_vars', default=['u', 'v', 'habitat', 'skipjack_cohort_20021015_density'],
+                   help='List of field variable names, as given in the NetCDF file. Order must match --variables args')
+    p.add_argument('-d', '--dimensions', default=['lat', 'lon', 'time'],
+                   help='List of PARCELS convention named dimensions across which field variables occur')
+    p.add_argument('-m', '--map_dimensions', default=['lat', 'lon', 'time'],
+                   help='List of dimensions across which field variables occur, as given in the NetCDF files, to map to the --dimensions args')
+    p.add_argument('-o', '--output_file', default='SEAPODYM_Particles',
+                   help='Name of output NetCDF file for particle data')
+    args = p.parse_args()
+
+    if args.location is None:
+        datadir = path.local().dirpath()
+        datadir = str(datadir) + '/' + 'Test_Data'
+    else:
+        datadir = args.location
+
+    # Build required dictionaries for NETCDF_Grid from cmd arguments
+    filenames = {}
+    for vars, var_files in zip(args.variables, args.files):
+        filenames.update({vars: (datadir + "/" + var_files)})
+    variables = {}
+    for vars, net_vars in zip(args.variables, args.netcdf_vars):
+        variables.update({vars: net_vars})
+    dimensions = {}
+    for dims, net_dims in zip(args.dimensions, args.map_dimensions):
+        dimensions.update({dims: net_dims})
+
+    grid = Grid.from_netcdf(filenames=filenames, variables=variables, dimensions=dimensions, vmax=5)
+
+    lat_min = min(grid.U.lat)
+    lat_max = max(grid.U.lat)
+    lon_min = min(grid.U.lon)
+    lon_max = max(grid.U.lon)
+
+    ParticleClass = JITParticle if args.mode == 'jit' else Particle
+
+    # First, we define a custom Particle class to which we add a
+    # custom variable, the habitat index H
+    class SEAPODYMParticle(ParticleClass):
+        # JIT compilation requires a-priori knowledge of the particle
+        # data structure, so we define additional variables here.
+        user_vars = {'habitat': np.float32}
+
+        def __init__(self, *args, **kwargs):
+            """Custom initialisation function which calls the base
+            initialisation and adds the instance variable p"""
+            super(SEAPODYMParticle, self).__init__(*args, **kwargs)
+
+        def __repr__(self):
+            """Custom print function which overrides the built-in"""
+            return "P(%.4f, %.4f)[H=%.5f]" % (self.lon, self.lat,
+                                              self.habitat)
+
+    pset = grid.ParticleSet(size=args.particles, pclass=Particle,
+                            start=((lon_min + lon_max) / 2, lat_min), finish=((lon_min + lon_max) / 2, lat_max))
+
+    hours = 3 * 24
+    substeps = 30
+
+    k_adv = pset.Kernel(AdvectionRK4)
+    sense_habitat = pset.Kernel(updateUserVars)
```

Comment:
Actually, it would have been better to have not commit the test script on
my part, as it was more a working piece of test code and as you note now-
some of the functions I'm not really running. Similarly with the
updateUserVars kernel function, I kept this in the test script until we
were happy with the way I'd put the functionality in particle.py.

Perhaps I should instead write a more generic test based on peninsula or
moving eddies where we create an arbitrary grid to be sampled by the
particules rather than relying on the SEAPODYM data which does not really
need to be used as a test specifically.

On 8 March 2016 at 07:43, Michael Lange <notifications@github.com> wrote:

> In SEAPODYM_Data_Test.py
> <https://github.com/OceanPARCELS/parcels/pull/43#discussion_r55321787>:
>
> > +            initialisation and adds the instance variable p"""
> > +            super(SEAPODYMParticle, self).__init__(*args, **kwargs)
> > +
> > +        def __repr__(self):
> > +            """Custom print function which overrides the built-in"""
> > +            return "P(%.4f, %.4f)[H=%.5f]" % (self.lon, self.lat,
> > +                                              self.habitat)
> > +
> > +    pset = grid.ParticleSet(size=args.particles, pclass=Particle,
> > +                            start=((lon_min + lon_max) / 2, lat_min), finish=((lon_min + lon_max) / 2, lat_max))
> > +
> > +    hours = 3 * 24
> > +    substeps = 30
> > +
> > +    k_adv = pset.Kernel(AdvectionRK4)
> > +    sense_habitat = pset.Kernel(updateUserVars)
>
> I don't think sense_habitat is run anywhere?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OceanPARCELS/parcels/pull/43/files#r55321787>.
>